### PR TITLE
Add stac-geoparquet and stac-arrow

### LINKED ISSUE #269: Add stac-geoparquet reading and writing

### PR DESCRIPTION
**Pull request by gadomski**
## Closes

- Supersedes #256 
- Closes #269

## Description

A re-start of the arrow/parquet work. **geoarrow-rs** seems to be in a bit of flux (especially w.r.t. 2 vs. 3 dimensions) so we're on a sorta-random SHA right now there.

## Checklist

- [x] Unit tests
- [x] Documentation, including doctests
- [x] Git history is linear
- [x] Commit messages are descriptive
- [x] (optional) Git commit messages follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Code is formatted (`cargo fmt`)
- [x] `cargo test`
- [x] Changes are added to the CHANGELOG

### CODE REVIEW

**Comment by kylebarron** [comment]
Did you hit specific issues with latest main of geoarrow-rs, or just that it requires some changes to your code?

File: stac-arrow/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+[![GitHub Workflow Status](https://img.shields.io/github/actions/workflow/status/stac-utils/stac-rs/ci.yml?branch=main&style=for-the-badge)](https://github.com/stac-utils/stac-rs/actions/workflows/ci.yml)
+[![docs.rs](https://img.shields.io/docsrs/stac-arrow?style=for-the-badge)](https://docs.rs/stac-arrow/latest/stac_arrow/)
+[![Crates.io](https://img.shields.io/crates/v/stac-arrow?style=for-the-badge)](https://crates.io/crates/stac-arrow)
+![Crates.io](https://img.shields.io/crates/l/stac-arrow?style=for-the-badge)
+[![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-2.1-4baaaa.svg?style=for-the-badge)](./CODE_OF_CONDUCT)
+
+Convert STAC item collections to and from [geoarrow](https://github.com/geoarrow/geoarrow-rs/) tables.
+To read and write [stac-geoparquet](https://github.com/stac-utils/stac-geoparquet), use [our crate with the same name](../stac-geoparquet/).
+
+**WARNING**: This library should be considered experimental while [geoarrow-rs](https://github.com/geoarrow/geoarrow-rs/) stabalizes.
```
Comment (kylebarron) [inline stac-arrow/README.md:12]:
```suggestion
**WARNING**: This library should be considered experimental while [geoarrow-rs](https://github.com/geoarrow/geoarrow-rs/) stabilizes.
```

**Comment by gadomski** [comment]
> Did you hit specific issues with latest main of geoarrow-rs, or just that it requires some changes to your code?

Yeah, I hadn't opened an issue yet, but it looks like the `Dimension` hash set isn't being updated in https://github.com/geoarrow/geoarrow-rs/blob/c927f1281bf9de028687733c5013e5246faef4e5/src/datatypes.rs#L556-L668

**Comment by kylebarron** [comment]
> Yeah, I hadn't opened an issue yet, but it looks like the `Dimension` hash set isn't being updated in [geoarrow/geoarrow-rs@`c927f12`/src/datatypes.rs#L556-L668](https://github.com/geoarrow/geoarrow-rs/blob/c927f1281bf9de028687733c5013e5246faef4e5/src/datatypes.rs#L556-L668)

Ah yes good catch. If you can create an issue to track that would be helpful

**Comment by gadomski** [comment]
Done, https://github.com/geoarrow/geoarrow-rs/issues/682

**Pull request merged**